scipy/signal/signaltools.py

```diff
@@ -2713,24 +2715,43 @@ def decimate(x, q, n=None, ftype='iir', axis=-1):
     --------
     resample
 
+    Notes
+    -----
+    The ``zero_phase`` keyword was added in 0.18.0.
+    The possibility to use instances of ``lti`` as ``ftype`` was added in
+    0.18.0.
     """
 
     if not isinstance(q, int):
         raise TypeError("q must be an integer")
 
-    if n is None:
-        if ftype == 'fir':
+    if ftype == 'fir':
+        if n is None:
             n = 30
-        else:
-            n = 8
+        system = lti(firwin(n + 1, 1. / q, window='hamming'), 1.)
 
-    if ftype == 'fir':
-        b = firwin(n + 1, 1. / q, window='hamming')
-        a = 1.
+    elif ftype == 'iir':
+        if n is None:
+            n = 8
+        system = lti(*cheby1(n, 0.05, 0.8 / q))
+    elif isinstance(ftype, lti):
+        system = ftype
     else:
-        b, a = cheby1(n, 0.05, 0.8 / q)
-
-    y = lfilter(b, a, x, axis=axis)
+        raise ValueError('invalid ftype')
+
+    if zero_phase is None:
+        warnings.warn("""
+                      Note: zero_phase will default to True in scipy v0.20.
+                      Until then, decimate defaults to one-way filtering for
+                      backwards compatibility. Ideally, always set this kwarg
+                      explicitly.
+                      """)
```

Comment:
Can you make this a ``FutureWarning``? 

And better make it simply "in a future release" rather than v0.20. Because timing of making changes varies, and because I'd like to never get to v0.20 (but 1.0 before that).